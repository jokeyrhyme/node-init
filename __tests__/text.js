'use strict';

const { withBadge, withLine } = require('../lib/text.js');

test('withBadge', () => {
  expect(withBadge('', '[badge](https://github.com)')).toMatchSnapshot();
  expect(withBadge('# name', '[badge](https://github.com)')).toMatchSnapshot();
});

test('withLine()', () => {
  const text = `
checkexisting
  checkwhitespace
check-hyphen
`;

  expect(withLine(text, 'checkexisting')).toMatchSnapshot();
  expect(withLine(text, 'checkwhitespace')).toMatchSnapshot();
  expect(withLine(text, 'check-hyphen')).toMatchSnapshot();
  expect(withLine(text, 'checkmissing')).toMatchSnapshot();
});
