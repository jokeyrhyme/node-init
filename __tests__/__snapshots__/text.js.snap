// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withBadge 1`] = `" [badge](https://github.com)"`;

exports[`withBadge 2`] = `"# name [badge](https://github.com)"`;

exports[`withLine() 1`] = `
"
checkexisting
  checkwhitespace
check-hyphen
"
`;

exports[`withLine() 2`] = `
"
checkexisting
  checkwhitespace
check-hyphen
"
`;

exports[`withLine() 3`] = `
"
checkexisting
  checkwhitespace
check-hyphen
"
`;

exports[`withLine() 4`] = `
"
checkexisting
  checkwhitespace
check-hyphen

checkmissing
"
`;
